### ISSUE TITLE
Clang warnings with SetUp methods in unit test fixtures

### ISSUE THREAD

**Issue by sayerhs** [NaluCFD/Nalu#280]
Clang on OS X raises overloaded virtual warnings with the following code snippet:

```
Nalu/unit_tests/UnitTestUtils.h:216:7: warning:
      'ABLWallFunctionHex8ElementWithBCFields::SetUp' hides overloaded virtual
      function [-Woverloaded-virtual]
 void SetUp(const double &rho, const double &utau, const double up, cons...
      ^
/Users/sanantha/Documents/nrel/code/nalu/install/include/yaml-cpp/../gtest/gtest.h:430:16: note:
      hidden overloaded virtual function 'testing::Test::SetUp' declared here:
      different number of parameters (0 vs 4)
  virtual void SetUp();
               ^
```

Also can this fixture be isolated into its own header file? `UnitTestUtils` gets included in almost every unit test C file.

**Comment by mbarone81** [comment]
@sayerhs or @rcknaus can one of you help with the clang warnings?  I do not build on OS X,

**Comment by sayerhs** [comment]
You can quell these warnings with the following modification: 

```
diff --git a/unit_tests/UnitTestUtils.h b/unit_tests/UnitTestUtils.h
index 9670cf3..36cc9b3 100644
--- a/unit_tests/UnitTestUtils.h
+++ b/unit_tests/UnitTestUtils.h
@@ -213,6 +213,8 @@ class ABLWallFunctionHex8ElementWithBCFields : public Hex8ElementWithBCFields
     ypSpec_(0.25)
  {}

+ using ::testing::Test::SetUp;
+
  void SetUp(const double &rho, const double &utau, const double up, const double yp)
  {
    rhoSpec_ = rho;
```

**Comment by mbarone81** [comment]
@shreyas -- thanks, created a pull request for quick fix for clang warnings

**Comment by mbarone81** [comment]
Separating out these classes into a different header file is probably good -- however, we could do this for other classes in UnitTestUtils as well, so maybe a more comprehensive review of the code structure should be done.

**Comment by mbarone81** [comment]
@sayerhs release and debug both build fine on gcc4.9.3 and intel-17

**Comment by sayerhs** [comment]
Fixed in #281
